opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1136,15 +1158,18 @@ namespace Opm {
                 for (int c = 0; c < nc; ++c) {
                     const int region = fipnum[c] - 1;
                     if (region != -1 && mask[c]) {
-                        fip_.fip[FIPData::FIP_PV][c] = pv[c];
                         const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
                         const auto& fs = intQuants.fluidState();
                         const double hydrocarbon = fs.saturation(FluidSystem::oilPhaseIdx).value() + fs.saturation(FluidSystem::gasPhaseIdx).value();
+                        const double pv =
+                            ebosSimulator_.model().dofTotalVolume(c)
+                            * ebosSimulator_.problem().porosity(c);
+                        fip_.fip[FIPData::FIP_PV][c] = pv;
```

Comment:
depends on what you mean with *calculated* since the `.model().dofTotalVolume(c)` and `problem().porosity(c)` methods are trivial. I'd say the pore volume was *accessed* three times here.